contrib/OtlpGrpc/Exporter.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\OtlpGrpc;
+
+use grpc;
+use Opentelemetry\Proto;
+use Opentelemetry\Proto\Collector\Trace\V1;
+use Opentelemetry\Proto\Common\V1\InstrumentationLibrary;
+use Opentelemetry\Proto\Trace\V1\InstrumentationLibrarySpans;
+use OpenTelemetry\Sdk\Trace;
+use OpenTelemetry\Trace as API;
+
+use OpenTelemetry\Trace\Span;
+
+class Exporter implements Trace\Exporter
+{
+    /**
+     * @var string
+     */
+    private $endpointURL;
+
+    /**
+     * @var string
+     */
+    private $protocol;
+
+    /**
+     * @var string
+     */
+    private $insecure;
+
+    /**
+     * @var string
+     */
+    private $certificateFile;
+
+    /**
+     * @var array
+     */
+    private $headers;
+
+    /**
+     * @var string
+     */
+    private $compression;
+
+    /**
+     * @var int
+     */
+    private $timeout;
+    /**
+     * @var SpanConverter
+     */
+    private $spanConverter;
+    
+    /**
+    * @var bool
+    */
+    private $running = true;
+
+    /**
+     * @var ClientInterface
+     */
+
+    private $client;
+
+    /**
+     * OTLP GRPC Exporter Constructor
+     * @param string $serviceName
+     */
+    public function __construct(
+        string $endpointURL = 'localhost:4317',
+        bool $insecure = true,
+        string $certificateFile = null,
+        array $headers = [],
+        bool $compression = false,
+        int $timeout = 10,
+        ClientInterface $client = null
+    ) {
+
+        // Set default values based on presence of env variable
+        $this->endpointURL = getenv('OTEL_EXPORTER_OTLP_ENDPOINT') ?: $endpointURL;
+        $this->protocol = getenv('OTEL_EXPORTER_OTLP_PROTOCOL') ?: 'grpc'; // I guess this is redundant?
+        $this->insecure = getenv('OTEL_EXPORTER_OTLP_INSECURE') ?: $insecure;
+        $this->certificateFile = getenv('OTEL_EXPORTER_OTLP_CERTIFICATE') ?: $certificateFile;
+        $this->headers[] = getenv('OTEL_EXPORTER_OTLP_HEADERS') ?: $headers; // TODO
+        $this->compression = getenv('OTEL_EXPORTER_OTLP_COMPRESSION') ?: $compression; // TODO
+        $this->timeout =(int) getenv('OTEL_EXPORTER_OTLP_TIMEOUT') ?: $timeout; // TODO
+
+        $this->spanConverter = new SpanConverter('foo');
+
+        if (!$this->insecure && !$this->certificateFile) {
+            // Assumed default
+            $_credentials = Grpc\ChannelCredentials::createSsl();
+        } elseif (!$this->insecure && $this->certificateFile) {
+            // Should we validate more?
+            $_credentials = Grpc\ChannelCredentials::createSsl(file_get_contents($certificateFile));
+        } else {
+            $_credentials = Grpc\ChannelCredentials::createInsecure();
+        }
+
+        $opts = [
+            'credentials' => $_credentials,
+            'update_metadata' =>  function () {
```

Comment:
I'd suggest either passing the metadata in via the constructor or having a function that sets it.  Then your exporter can work with ALL backends :) because this exporter is *fantastic* and is able to send to other backends.

I commented out these lines metadata lines, and was able to send to another backend:
![image](https://user-images.githubusercontent.com/43715151/111918206-3fa1ae80-8a41-11eb-8ed2-d83a52ae9fd1.png)

Reply:
Thanks, yeah these should be passed via the constructor. I had attempted it, but I got stuck trying to support passing an array via the env vars: https://github.com/SeanHood/opentelemetry-php/blob/b8469e693ba698f2bf3bb9d0ccc78a66ccc69375/contrib/OtlpGrpc/Exporter.php#L88

I think I need to use `explode()`, otel-python and otel-ruby support passing these like `x-header-1=foo,x-header-2=bar`. I think this format should also work in otel-php.

Reply:
Okay, I figured this out. Will commit later + working on Exporter tests

Reply:
Passing these headers in string format is now implemented